pgjdbc/src/test/java/org/postgresql/test/jdbc2/QuotationTest.java

```diff
@@ -0,0 +1,131 @@
+package org.postgresql.test.jdbc2;
+
+import org.postgresql.test.TestUtil;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collection;
+
+@RunWith(Parameterized.class)
+public class QuotationTest extends BaseTest4 {
+  enum QuoteStyle {
+    SIMPLE("'"), DOLLAR_NOTAG("$$"), DOLLAR_A("$a$"), DOLLAR_DEF("$DEF$");
+
+    private final String quote;
+
+    QuoteStyle(String quote) {
+      this.quote = quote;
+    }
+
+    @Override
+    public String toString() {
+      return quote;
+    }
+  }
+
+  private final QuoteStyle quoteStyle;
+  private final String expr;
+  private final String[] args;
+  private final String expected;
+
+  public QuotationTest(QuoteStyle quoteStyle, String expected, String expr, String[] args) {
```

Comment:
Nice catch. Initially I thought to use it as "bind arguments", then it turned out to be unused.